### PR TITLE
Update whiptails

### PR DESCRIPTION
**Pull request by szaimen**
fix https://github.com/nextcloud/vm/issues/1058

### CODE REVIEW

**Comment by szaimen** [comment]
@enoch85 this should be all places where to fix the whiptails.
I've checked the list.
Other places are already correct.

File: apps/additional_apps.sh [hunk 1]
```diff
@@ -28,76 +28,72 @@ whiptail --title "Which apps do you want to install?" --checklist --separate-out
...
+    *"Adminer"*)
+        clear
+        run_app_script adminer
+    ;;&
+
+    *"Netdata"*)
+        clear
+        run_app_script netdata
+    ;;&
+
```
Comment (szaimen) [inline apps/additional_apps.sh]: What would you say? Shall we delete the empty rows here?

File: apps/previewgenerator.sh [hunk 2]
```diff
@@ -107,60 +107,56 @@ then
...
+        ;;&
+
+        *"MarkDown"*)
+            occ_command config:system:set enabledPreviewProviders 4 --value="OC\\Preview\\MarkDown"
+        ;;&
+
+        *"MP3"*)
+            occ_command config:system:set enabledPreviewProviders 5 --value="OC\\Preview\\MP3"
+        ;;&
+
```
Comment (szaimen) [inline apps/previewgenerator.sh]: Or what about here?

File: nextcloud_install_production.sh [hunk 3]
```diff
@@ -668,65 +666,61 @@ a2ensite "$HTTP_CONF"
...
+    ;;&
+    *"Contacts"*)
+        install_and_enable_app contacts
+    ;;&
+    *"IssueTemplate"*)
+        install_and_enable_app issuetemplate
+    ;;&
+    *"PDFViewer"*)
+        install_and_enable_app files_pdfviewer
+    ;;&
```
Comment (szaimen) [inline nextcloud_install_production.sh]: This is how I would do it.
(Without empty lines)

File: nextcloud_install_production.sh [hunk 4]
```diff
@@ -157,51 +157,49 @@ You could still choose to only run on one disk though, which is not recommended,
...
-        "1 Disk")
-            print_text_in_color "$IRed" "1 Disk setup chosen."
-            sleep 2
-        ;;
-        *)
-        ;;
-    esac
+choice=$(whiptail --title "Choose disk format" --radiolist "How would you like to configure your disks?\nSelect by pressing the spacebar and ENTER" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+"2 Disks Auto" "(Automatically configured)            " ON \
+"2 Disks Manual" "(Choose by yourself)            " OFF \
```
Comment (szaimen) [inline nextcloud_install_production.sh]: And what about the spaces here?

File: nextcloud_install_production.sh [hunk 5]
```diff
@@ -157,51 +157,49 @@ You could still choose to only run on one disk though, which is not recommended,
...
-            sed -i "s|#DNS=.*|DNS=$GATEWAY|g" /etc/systemd/resolved.conf
-            sed -i "s|#FallbackDNS=.*|FallbackDNS=149.112.112.112 2620:fe::9|g" /etc/systemd/resolved.conf
-        ;;
-        *)
-        ;;
-    esac
+choice=$(whiptail --title "Set DNS Resolver" --radiolist "Which DNS provider should this Nextcloud box use?\nSelect by pressing the spacebar and ENTER" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+"Quad9" "(https://www.quad9.net/)            " ON \
+"Cloudflare" "(https://www.cloudflare.com/dns/)            " OFF \
+"Local" "($GATEWAY + 149.112.112.112)              " OFF 3>&1 1>&2 2>&3)
```
Comment (szaimen) [inline nextcloud_install_production.sh]: And here?

File: nextcloud_install_production.sh [hunk 6]
```diff
@@ -668,65 +666,61 @@ a2ensite "$HTTP_CONF"
...
-rm -f results
+choice=$(whiptail --title "Install apps or software" --checklist "Automatically configure and install selected apps or software\nDeselect by pressing the spacebar" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+"Calendar" "            " ON \
+"Contacts" "            " ON \
+"IssueTemplate" "       " ON \
+"PDFViewer" "           " ON \
+"Extract" "             " ON \
+"Text" "                " ON \
+"Mail" "                " ON \
+"Deck" "                " ON \
```
Comment (szaimen) [inline nextcloud_install_production.sh]: And also here?

Comment (szaimen) [reply nextcloud_install_production.sh]: I honestly woud remove them.

File: apps/additional_apps.sh (hunk 1 above)
Comment (enoch85) [reply apps/additional_apps.sh]: Are you really working with the latest master? 

Yes please, delete emtpy lines.

**Comment by enoch85** [comment]
-[ ] apps/smbmount.sh
-[ ] apps/previewgenerator.sh
-[ ] apps/additional_apps.sh
-[ ] static/cookielifetime.sh
-[ ] nextcloud_install_production.sh
-[ ] nextcloud-startup-script.sh

File: apps/additional_apps.sh (hunk 1 above)
Comment (szaimen) [reply apps/additional_apps.sh]: @enoch85 Yes, I am working with the latest master afaik.
and why is this resolved?

**Comment by szaimen** [comment]
@enoch85 I have removed all empty lines and unnecessary spaces. This is ready imo.

File: apps/additional_apps.sh (hunk 1 above)
Comment (enoch85) [reply apps/additional_apps.sh]: >  I have removed all empty lines and unnecessary spaces

File: nextcloud_install_production.sh [hunk 7]
```diff
@@ -157,51 +157,49 @@ You could still choose to only run on one disk though, which is not recommended,
...
-        *)
-        ;;
-    esac
+choice=$(whiptail --title "Set DNS Resolver" --radiolist "Which DNS provider should this Nextcloud box use?\nSelect by pressing the spacebar and ENTER" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+"Quad9" "(https://www.quad9.net/)" ON \
+"Cloudflare" "(https://www.cloudflare.com/dns/)" OFF \
+"Local" "($GATEWAY + 149.112.112.112)" OFF 3>&1 1>&2 2>&3)
+
+case "$choice" in
+    Quad9)
```
Comment (enoch85) [inline nextcloud_install_production.sh]: There are no quotes here. Please fix.

**Comment by enoch85** [comment]
Tested `additional_apps.sh` and it worked. When I ran with `set -x` I understand how it works.

**Comment by szaimen** [comment]
What is set -x ?

File: nextcloud_install_production.sh (hunk 7 above)
Comment (szaimen) [reply nextcloud_install_production.sh]: You don't need the quotes here since it is just one word without spaces.

**Comment by enoch85** [comment]
try `bash -x script.sh`

File: nextcloud_install_production.sh (hunk 7 above)
Comment (enoch85) [reply nextcloud_install_production.sh]: ! @szaimen

Comment (szaimen) [reply nextcloud_install_production.sh]: Okay, please wait

Comment (enoch85) [reply nextcloud_install_production.sh]: The thing with a "standard" is that it should be the same everywhere.

Like
https://github.com/nextcloud/vm/pull/1071/files#diff-11ba175bd07d41f8a89a19da883fff86R21-R30
https://github.com/nextcloud/vm/pull/1071/files#diff-9fd0b2a4bd9795c3884c76936a9f11c0R113-R143
https://github.com/nextcloud/vm/pull/1071/files#diff-9fd0b2a4bd9795c3884c76936a9f11c0R173-R194

And so on... :)

Comment (szaimen) [reply nextcloud_install_production.sh]: I will do that. Please be patient!

**Comment by szaimen** [comment]
> try `bash -x script.sh`

okay, seems to be the same like setting the debug option at the start of the script to 1...
But its too fast to read in real time.

File: nextcloud_install_production.sh (hunk 7 above)
Comment (szaimen) [reply nextcloud_install_production.sh]: This is done 👍

**Comment by szaimen** [comment]
@enoch85 the detail is fixed.

File: nextcloud-startup-script.sh [hunk 8]
```diff
@@ -374,41 +374,33 @@ sleep 3
...
-            clear
-            run_static_script cookielifetime
-        ;;
-
-        *)
-        ;;
-    esac
-done 4< results
-rm -f results
+case *"$choice" in
```
Comment (szaimen) [inline nextcloud-startup-script.sh]: @enoch85 I think here is a typo...

Comment (enoch85) [reply nextcloud-startup-script.sh]: Did you test, and it didn't work? If so, please comitt a PR.

Thanks! :+1:

Comment (szaimen) [reply nextcloud-startup-script.sh]: I have tested it and it works. So no need to fix that.

Comment (enoch85) [reply nextcloud-startup-script.sh]: doesn't it look the same everywhere else? It should be consistent IMO.

Comment (szaimen) [reply nextcloud-startup-script.sh]: no, this is different (was a mistake) I could open a PR to fix that, but this is a bit to less for a PR imo. I will put it in https://github.com/nextcloud/vm/pull/1072

Comment (szaimen) [reply nextcloud-startup-script.sh]: this is fixed here: https://github.com/nextcloud/vm/pull/1072/commits/4535ec6ef8376be9101e6abad68eb77f4681db41

**Pull request merged**